framework/lib/components/createable-select-dropdown/createable-select-dropdown.tsx

```diff
@@ -78,95 +79,98 @@ import { customComponents } from './custom-components'
   creationOption = {
-    label: 'Add option...',
-    value: 'Add option...',
+    label: 'Add field',
+    value: 'add_field',
     isCreation: true,
   },
   onOptionChange,
-  width = '100%',
   variant = 'outline',
   defaultValue,
   value,
   menuPlacement = 'bottom',
 }: CreatableSelectDropdownProps<T>) => {
-  const initialSelectedOption = useMemo(() => {
-    if (!isNil(value)) {
-      return standardOptions.find((option) => option.value === value) ?? null
-    } if (!isNil(defaultValue)) {
-      return standardOptions.find((option) => option.value === defaultValue) ?? null
-    }
-    return null
-  }, [ value ]
-  )
+  const [ selectedOption, setSelectedOption ] = useState<Option | null>(() => {
+    const targetValue = value ?? defaultValue ?? null
+    return targetValue
+      ? standardOptions.find((option) => option.value === targetValue) ?? null
+      : null
+  })
 
-  const [ selectedOption, setSelectedOption ] =
-    useState<Option | null>(initialSelectedOption)
-  const [ newOptionText, setNewOptionText ] = useState('')
-  const [ newOptionPlaceholder, setNewOptionPlaceholder ] = useState(initialPlaceholder)
+  const [ newOptionPlaceholder, setNewOptionPlaceholder ] =
+    useState(initialPlaceholder)
   const [ createdOptions, setCreatedOptions ] = useState<Option[]>([])
+  const [ createNewOption, setCreateNewOption ] = useState<boolean>(false)
+  const [ addFieldInputValue, setAddFieldInputValue ] = useState<string>('')
+  const CREATION_OPTION_VALUE = 'add_field'
 
   function isCreationOption (option: any): option is CreationOption {
     return option && typeof option.isCreation === 'boolean'
   }
 
   const ref = useRef<HTMLDivElement>(null)
-  useOutsideClick({
-    ref,
-    handler: () => {
-      if (isCreationOption(selectedOption)) {
-        setSelectedOption(initialSelectedOption)
-        setNewOptionPlaceholder(initialPlaceholder)
-      }
-    },
-  })
 
-  const handleInputChange = (newValue: string) => {
-    setNewOptionText(newValue)
-  }
+  const handleCreateOption = (newOptionTextString: string) => {
+    setCreateNewOption(true)
 
-  const handleCreateOption = () => {
-    const newOptionTextLower = newOptionText.toLowerCase()
-    const newOptionExists = standardOptions.some((option) =>
-      option.value.toLowerCase() === newOptionTextLower)
+    if (isEmpty(trim(newOptionTextString))) {
+      return
+    }
+
+    const newOptionTextLower = toLower(newOptionTextString)
+
+    const newOptionExistsInStandardOptions = any(
+      (option) => toLower(option.value) === newOptionTextLower,
+      standardOptions
+    )
 
-    if (!newOptionExists) {
-      const newOption = { label: newOptionText, value: newOptionText }
-      setCreatedOptions((currentCreatedOptions) => [ ...currentCreatedOptions, newOption ])
+    const newOptionExistsInCreatedOptions = any(
+      (option) => toLower(option.value) === newOptionTextLower,
+      createdOptions
+    )
```

Comment:
last comment. remove duplication by moving "any" functionality to a separate function that accepts `standardOptions/createdOptions`.